### ISSUE TITLE
"Windows x86 PE RTTI Analyzer" OOB

### ISSUE THREAD

**Issue by Arcnor** [NationalSecurityAgency/ghidra#2153]
**Describe the bug**
While running the "Windows x86 PE RTTI Analyzer" I'm getting an OOB exception on Ghidra 9.2 (today's master, a77dcf7), tried with many different PE files. This works fine on 9.1.2 PUBLIC.

```
Analysis Task: Windows x86 PE RTTI Analyzer - Index 0 out of bounds for length 0
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248)
	at java.base/java.util.Objects.checkIndex(Objects.java:372)
	at java.base/java.util.ArrayList.get(ArrayList.java:458)
	at ghidra.app.cmd.data.rtti.CreateRtti4BackgroundCmd.<init>(CreateRtti4BackgroundCmd.java:67)
	at ghidra.app.plugin.prototype.MicrosoftCodeAnalyzerPlugin.RttiAnalyzer.processRtti4sForRtti0(RttiAnalyzer.java:182)
	at ghidra.app.plugin.prototype.MicrosoftCodeAnalyzerPlugin.RttiAnalyzer.processRtti0(RttiAnalyzer.java:166)
	at ghidra.app.plugin.prototype.MicrosoftCodeAnalyzerPlugin.RttiAnalyzer.added(RttiAnalyzer.java:120)
	at ghidra.app.plugin.core.analysis.AnalysisScheduler.runAnalyzer(AnalysisScheduler.java:190)
	at ghidra.app.plugin.core.analysis.AnalysisTask.applyTo(AnalysisTask.java:39)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager$AnalysisTaskWrapper.run(AutoAnalysisManager.java:685)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:785)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:664)
	at ghidra.app.plugin.core.analysis.AutoAnalysisManager.startAnalysis(AutoAnalysisManager.java:629)
	at ghidra.app.plugin.core.analysis.AnalysisBackgroundCommand.applyTo(AnalysisBackgroundCommand.java:58)
	at ghidra.framework.plugintool.mgr.BackgroundCommandTask.run(BackgroundCommandTask.java:101)
	at ghidra.framework.plugintool.mgr.ToolTaskManager.run(ToolTaskManager.java:315)
	at java.base/java.lang.Thread.run(Thread.java:834)

---------------------------------------------------
Build Date: 2020-Jul-31 1051 WEST
Ghidra Version: 9.2
Java Home: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
JVM Version: AdoptOpenJDK 11.0.7
OS: Mac OS X 10.14.6 x86_64
```

**Comment by emteere** [comment]
Is this on a from scratch import, or a file you had previously imported in 9.1.2?

**Comment by Arcnor** [comment]
The DB existed, but there was no analysis on it. I also tried on a file already analyzed, but same result.

**Comment by emteere** [comment]
I've fixed the issue in a branch, it may take a day or so to make it to the github repository.
Is there a binary perhaps in windows that exhibited this problem?

In the meantime, you can replace the following lines in the RttiAnalyzer:181
```
		// create all found RTTI4 tables at once
		if (rtti4Addresses.size() > 0) {
			CreateRtti4BackgroundCmd cmd = new CreateRtti4BackgroundCmd(rtti4Addresses, dataBlocks,
				validationOptions, applyOptions);
			cmd.applyTo(program, monitor);
```
		}

**Comment by ghizard** [comment]
Should be addressed by c6d3a7d
If not, please reopen.
